utils/crypto_dir.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/openshift-kni/lifecycle-agent/api/seedreconfig"
+	"github.com/openshift-kni/lifecycle-agent/internal/common"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	cryptoDirMode = 0o600
+)
+
+func SeedReconfigurationKubeconfigRetentionToCryptoDir(cryptoDir string, kubeconfigCryptoRetention *seedreconfig.KubeConfigCryptoRetention) error {
+	if err := os.MkdirAll(cryptoDir, os.ModePerm); err != nil {
+		return fmt.Errorf("error creating %s: %w", cryptoDir, err)
+	}
+
+	if err := os.WriteFile(path.Join(cryptoDir, "admin-kubeconfig-client-ca.crt"),
+		[]byte(kubeconfigCryptoRetention.KubeAPICrypto.ClientAuthCrypto.AdminCACertificate), cryptoDirMode); err != nil {
+		return fmt.Errorf("error writing admin-kubeconfig-client-ca.crt: %w", err)
+	}
+
+	if err := os.WriteFile(path.Join(cryptoDir, "loadbalancer-serving-signer.key"),
+		[]byte(kubeconfigCryptoRetention.KubeAPICrypto.ServingCrypto.LoadbalancerSignerPrivateKey), cryptoDirMode); err != nil {
+		return fmt.Errorf("error writing loadbalancer-serving-signer.key: %w", err)
+	}
+
+	if err := os.WriteFile(path.Join(cryptoDir, "localhost-serving-signer.key"),
+		[]byte(kubeconfigCryptoRetention.KubeAPICrypto.ServingCrypto.LocalhostSignerPrivateKey), cryptoDirMode); err != nil {
+		return fmt.Errorf("error writing localhost-serving-signer.key: %w", err)
+	}
+
+	if err := os.WriteFile(path.Join(cryptoDir, "service-network-serving-signer.key"),
+		[]byte(kubeconfigCryptoRetention.KubeAPICrypto.ServingCrypto.ServiceNetworkSignerPrivateKey), cryptoDirMode); err != nil {
+		return fmt.Errorf("error writing service-network-serving-signer.key: %w", err)
+	}
+
+	if err := os.WriteFile(path.Join(cryptoDir, "ingresskey-ingress-operator.key"),
+		[]byte(kubeconfigCryptoRetention.IngresssCrypto.IngressCA), cryptoDirMode); err != nil {
+		return fmt.Errorf("error writing ingresskey-ingress-operator.key: %w", err)
+	}
```

Comment:
Consider, changing it to a loop, eg:
for _, key := range []keyInfo{
{Name: "ingresskey-ingress-operator.key",
 key: kubeconfigCryptoRetention.IngresssCrypto.IngressCA},
{Name: "service-network-serving-signer.key",
 key: kubeconfigCryptoRetention.KubeAPICrypto.ServingCrypto.ServiceNetworkSignerPrivateKey},...
 }